code/modules/mining/lavaland/loot/legion_loot.dm

```diff
@@ -39,16 +50,81 @@
+			var/old_color = user.color
+			user.color = list(340/255, 240/255, 0,0, 0,0,0,0, 0,0,0,0, 0,0,0,1, 0,0,0,0)
+			var/old_transform = user.transform
+			user.transform *= 1.2
+			animate(user, color = old_color, transform = old_transform, time = 1 SECONDS)
+
+/obj/item/storm_staff/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(!thunder_charges)
+		to_chat(user, "<span class='warning'>The staff needs to recharge.</span>")
+		return
+	var/turf/target_turf = get_turf(target)
+	var/area/target_area = get_area(target)
+	var/area/user_area = get_area(user)
+	if(!target_turf || !target_area || (is_type_in_list(target_area, excluded_areas)) || !user_area || (is_type_in_list(user_area, excluded_areas)))
+		to_chat(user, "<span class='warning'>The staff will not work here.</span>")
+		return
+	if(target_turf in targeted_turfs)
+		to_chat(user, "<span class='warning'>That turf is already being shocked!</span>")
+		return
+	if(HAS_TRAIT(user, TRAIT_PACIFISM))
+		to_chat(user, "<span class='warning'>You don't want to hurt anyone!</span>")
+		return
+	var/power_boosted = FALSE
+	for(var/V in SSweather.processing)
+		var/datum/weather/W = V
+		if((target_turf.z in W.impacted_z_levels) && W.area_type == target_area.type)
+			power_boosted = TRUE
+			break
+	playsound(src, 'sound/magic/lightningshock.ogg', 10, TRUE, extrarange = SILENCED_SOUND_EXTRARANGE, falloff_distance = 0)
+	targeted_turfs += target_turf
+	to_chat(user, "<span class='warning'>You aim at [target_turf]!</span>")
+	new /obj/effect/temp_visual/thunderbolt_targeting(target_turf)
+	addtimer(CALLBACK(src, PROC_REF(throw_thunderbolt), target_turf, power_boosted), 1.5 SECONDS)
+	thunder_charges--
+	addtimer(CALLBACK(src, PROC_REF(recharge)), thunder_charge_time)
+
+/obj/item/storm_staff/proc/recharge(mob/user)
+	thunder_charges = min(thunder_charges + 1, max_thunder_charges)
+	playsound(src, 'sound/magic/charge.ogg', 10, TRUE, extrarange = SILENCED_SOUND_EXTRARANGE, falloff_distance = 0)
+
+/obj/item/storm_staff/proc/throw_thunderbolt(turf/target, boosted)
+	targeted_turfs -= target
+	new /obj/effect/temp_visual/thunderbolt(target)
+	var/list/affected_turfs = list(target)
+	if(boosted)
+		for(var/direction in GLOB.alldirs)
+			var/turf_to_add = get_step(target, direction)
+			if(!turf_to_add)
+				continue
+			affected_turfs += turf_to_add
+	for(var/turf/turf as anything in affected_turfs)
+		new /obj/effect/temp_visual/electricity(turf)
+		for(var/mob/living/hit_mob in turf)
+			to_chat(hit_mob, "<span class='userdanger'>You've been struck by lightning!</span>")
+			hit_mob.electrocute_act(15 * (isanimal(hit_mob) ? 3 : 1) * (turf == target ? 2 : 1) * (boosted ? 2 : 1), src, flags = SHOCK_TESLA|SHOCK_NOSTUN)
+
+		for(var/obj/hit_thing in turf)
+			hit_thing.take_damage(20, BURN, ENERGY, FALSE)
+	playsound(target, 'sound/magic/lightningbolt.ogg', 100, TRUE)
+	target.visible_message("<span class='userdanger'>A thunderbolt strikes [target]!</span>")
```

Comment:
I don't think this should necessarily be userdanger since it's sent to anyone who can see the turf
```suggestion
	target.visible_message("<span class='danger'>A thunderbolt strikes [target]!</span>")
```

Reply:
done